pkg/autoscaler/statserver/server.go

```diff
@@ -154,17 +155,26 @@ func (s *Server) Handler(w http.ResponseWriter, r *http.Request) {
 			close(handlerCh)
 			return
 		}
-		if messageType != websocket.BinaryMessage {
-			s.logger.Error("Dropping non-binary message.")
+
+		// we accept either GOB-encoded or JSON-encoded messages depending on the
+		// message type to ensure safe upgrades.
+		var dec decoder
+		switch messageType {
+		case websocket.BinaryMessage:
+			dec = gob.NewDecoder(bytes.NewBuffer(msg))
+		case websocket.TextMessage:
+			dec = json.NewDecoder(bytes.NewBuffer(msg))
+		default:
+			s.logger.Error("Dropping unknown message type.")
```

Comment:
Log the message type?

Reply:
we didn't bother before so I didn't here, unfortunately the message type is just an int, so I _think_ we can only log some random integer :/.